### ISSUE TITLE
ui5-multi-input: Provide capabilities to set aria-haspopup and aria-expanded attributes

### ISSUE THREAD

**Issue by zhangliangyi** [SAP/ui5-webcomponents#2988]
**Is your feature request related to a problem? Please describe.**
Once a multi-input component with value-help is defined, we cannot set the `aria-haspopup` and `aria-expanded` properties to indicate to current status of the component. As a result, the user may not know that the component will open a popup via keyboard action.

**Describe the solution you'd like**
Provide some interfaces that the developer can set `aria-haspopup` and `aria-expanded` manually, and in `ui5-input` component, it will use the customized value in the highest priority.

**Describe alternatives you've considered**
N/A

**Additional context**
![image](https://user-images.githubusercontent.com/1867889/112021102-e4200f80-8b6b-11eb-99ed-5061fb68579c.png)

Organization: (if applicable)
SAP SF

**Comment by ilhan007** [comment]
Hello @zhangliangyi 

The MultiInput (the Input) has "_inputAccInfo' property for rendering acc related attrs
{
"ariaDescribedBy":
"ariaInvalid":
"ariaHasPopup": 
"ariaAutoComplete":
"role":
"ariaOwns":
"ariaExpanded":
"ariaDescription":
"ariaLabel":
}

Would this work for you
```js
multiInputInstance["_inputAccInfo"] = Object.assign(multiInputInstance["_inputAccInfo"], {ariaHasPopup: true});
multiInputInstance["_inputAccInfo"] = Object.assign(multiInputInstance["_inputAccInfo"], {ariaExpanded: true});

// always merge your values with the current object, because if you use it like this
// multiInputInstance["_inputAccInfo"] = {ariaHasPopup: true} 
// you will remove the other built-in attrs, listed above
```

**Comment by zhangliangyi** [comment]
@ilhan007 
Thank you for the advice. I will try it first locally and get back to you asap.

**Comment by zhangliangyi** [comment]
@ilhan007 
I just tried your advice in my local. It works except I need to call the re-render function to refresh the multiInputInstance to apply the aria properties to the shadow DOM.
```javascript
const multiInputInstance = this.innerField;
multiInputInstance._inputAccInfo = {
  ...multiInputInstance._inputAccInfo,
  ariaHasPopup: true,
  ariaExpanded: false
};
await RenderScheduler.renderDeferred(multiInputInstance);
```
For now, it is acceptable for me but I wonder if there is any improvement on this in the future.

**Comment by ilhan007** [comment]
Hello @zhangliangyi 

Before I forward your request to the responsible team, I would like to make a short summary.

You are using the MultiInput component and rely on the "value-help-trigger" event to open a popup.
However, you miss the support for "aria-haspopup" and "aria-expanded" attributes as they can't be really set from outside (well they can be set, but the attrs need to be manually forwarded to the right places to take effect).

I can see that when "suggestions" slot is used, the "aria-haspopup" is set to "true", which is correct, did not see the "aria-expanded" set, not sure if it's necessary to have that one as well or not (we have to check that).

But from what I understood, even without suggestions, you need to control the "aria-haspopup" and "aria-expanded" attributes to correctly sync the component state, due to the use of the value-help-trigger event, which might open a popup.

Is that correct? Do you want to add something else? And, could you also point to an example if possible, where you have the MultiInput + value-help-trigger in use.


BR,
ilhan

**Comment by zhangliangyi** [comment]
@ilhan007 
Totally correct.

What I need is to manipulate the `aria-expanded` and `aria-haspopup` attributes in the multi-input component.

In my case, the user only can add the tokens from the popup opened by `value-help-trigger`, so I did not use `suggestion` attribute. In other words, user can only select things in the popup, any typing in the multi-input is not valid and would be ignored.

**Comment by ilhan007** [comment]
Hello @hristop, @nikolay-kolarov @SAP/ui5-webcomponents-topic-rl 
could you take a look of this feature request, related to the MultiInput component and its a11y support.

Summary:
The author needs the ability to set and control the aria-expanded and aria-haspopup attributes.
His use-case is "users add tokens from a popup, opened after **value-help-trigger** event is fired.

As far I can see when suggestions are used, aria-haspopup is set and updated correctly, but the author does not use suggestions at all. The tokens are added from a different popup that is opened on value-help-trigger.
For the **aria-expanded**, I don't even know if it makes sense here.


If you need more details,  please have a look at the discussion we had so far or continue the thread.

BR,
ilhan

**Comment by nikolay-kolarov** [comment]
Hi @zhangliangyi,

Will it be possible to provide a sample of what exactly you are doing?
We had many debates about the correct aria-haspopup usage in UI5 and I want to see the exact case, so that we can decide how to move this issue forward.

Thanks in advance!

Best Regards,
Nikolay

**Comment by nikolay-kolarov** [comment]
Hi,

I am closing this one, as it was in author action for quite some time and we need additional info to decide how to continue.
Please reopen if you still have interest in this and provide the needed details.

Best Regards,
Nikolay
